### PR TITLE
Added Bluray576p Quality

### PR DESCRIPTION
**Pull request by Voyex**
#### Database Migration
YES

#### Description
Added the Bluray576p resolution and added a database migration to add the quality to existing profiles (with only Any & SD being enabled).

#### Todos
- [ ] Tests
- [x] Wiki Updates - Updated [Qualities Defined](https://wiki.servarr.com/en/sonarr/settings#qualities-defined)

#### Issues Fixed or Closed by this PR

*  #6203

### CODE REVIEW

File: src/NzbDrone.Core/Datastore/Migration/201_add_bluray576_quality_in_profile.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                });
+
+                var newBlurayProfile = new ProfileItem201
+                {
+                    Quality = quality,
+                    Allowed = false
+                };
+
+                // The Bluray576p quality should be allowed in only profiles 'Any' and 'SD'
+                if (profile.Id == 1 || profile.Id == 2)
```
[the remark this answers is not in the document]
Comment (Voyex) [reply src/NzbDrone.Core/Datastore/Migration/201_add_bluray576_quality_in_profile.cs:107]: Thank you, I appreciate the input. I was unsure which would be safer.

File: src/NzbDrone.Core.Test/Qualities/QualityFixture.cs [hunk 2]
```diff
@@ -20,6 +20,7 @@ public class QualityFixture : CoreTest
                         new object[] { 4, Quality.HDTV720p },
                         new object[] { 5, Quality.WEBDL720p },
                         new object[] { 6, Quality.Bluray720p },
+                        new object[] { 6, Quality.Bluray720p },
```
Comment (thezoggy) [inline src/NzbDrone.Core.Test/Qualities/QualityFixture.cs:23]: typo?

File: src/NzbDrone.Core/Datastore/Migration/201_add_bluray576_quality_in_profile.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public int Id { get; set; }
+
+        public string Name { get; set; }
+
+        public int? Quality { get; set; }
+
+        public bool Allowed { get; set; }
+        public List<ProfileItem201> Items { get; set; }
+    }
```
Comment (markus101) [inline src/NzbDrone.Core/Datastore/Migration/201_add_bluray576_quality_in_profile.cs:48]:
```suggestion
        public string Name { get; set; }
        public int? Quality { get; set; }
        public bool Allowed { get; set; }
        public List<ProfileItem201> Items { get; set; }
    }
```

File: src/NzbDrone.Core/Datastore/Migration/201_add_bluray576_quality_in_profile.cs [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        }
+
+        public void Commit()
+        {
+            foreach (var profile in _changedProfiles)
+            {
+                using (var updateProfileCmd = _connection.CreateCommand())
+                {
+                    updateProfileCmd.Transaction = _transaction;
+                    updateProfileCmd.CommandText = "UPDATE QualityProfiles SET Name = ?, Cutoff = ?, Items = ? WHERE Id = ?";
```
Comment (markus101) [inline src/NzbDrone.Core/Datastore/Migration/201_add_bluray576_quality_in_profile.cs:73]: These queries are invalid for Postgres, you'll need to mirror the existing migrations and correctly quote the columns/table names.

File: src/NzbDrone.Core/Datastore/Migration/201_add_bluray576_quality_in_profile.cs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        }
+
+        private List<Profile201> GetProfiles()
+        {
+            var profiles = new List<Profile201>();
+
+            using (var getProfilesCmd = _connection.CreateCommand())
+            {
+                getProfilesCmd.Transaction = _transaction;
+                getProfilesCmd.CommandText = @"SELECT Id, Name, Cutoff, Items FROM QualityProfiles";
```
Comment (markus101) [inline src/NzbDrone.Core/Datastore/Migration/201_add_bluray576_quality_in_profile.cs:125]: Same here.

**Review by markus101 — CHANGES_REQUESTED** [review]

**Comment by markus101** [comment]
Closing based on the comments in the issue.

**Pull request closed without merging**